### DISCUSSION TITLE
Drift Dectection in helm controller

### DISCUSSION THREAD

**Discussion by azouz34200** [fluxcd/flux2#3998] [Q&A]
Hello,
I installed flux in openshift 4.8 using OLM however I can't find any information how to enable drift detection when using operator. Is there a possible configuration with the operatorConfig or by adding config options in the subcription?
Thank you for your answers

**Comment by kingdonb** [comment]
Greetings,

I was not aware of `operatorConfig` – it looks like it's possible to pass in environment variables this way

In a typical Flux installation, you have Flux managing itself and it's possible for Flux to patch its own definition in order to make configuration changes that should have a global scope, like the Helm Drift Detection feature flag that you are trying to enable.

https://fluxcd.io/flux/cheatsheets/bootstrap/#enable-helm-drift-detection

In the OLM installation, as you've no doubt understood, the Operator Lifecycle Manager handles all of this, and Flux does not manage itself. So the standard configuration options are not available. We don't have a good solution for this yet. It's not currently possible to do all Flux configuration using only environment variables, you do need to occasionally set some command line arguments as flags for the Flux controllers in order to effect global changes.

We don't really recommend using Operator Hub Flux for this reason, it's more flexible if you bootstrap Flux onto the cluster in the normal way and disregard Operator Hub. Then you can use advanced features that most OpenShift users will want, like soft multi-tenancy and default service accounts, (all of those are talked about on the bootstrap cheatsheet page that I linked above.)

This answer is somewhat unsatisfying, as we do want to support Operator Hub flux, but right now it falls behind upstream Flux and unfortunately it can't be recommended very highly if you need to customize Flux installation at all.

Another option if you weren't happy with Flux bootstrap for some reason (lots of people don't want to run another CLI for managing infrastructure) is the Terraform Provider, which has bootstrapping capabilities since a few months ago, and also can be customized with patches in the way that OpenShift's Operator Hub Flux cannot.
